src/wasm-linker.cpp

```diff
@@ -381,12 +381,14 @@ void Linker::makeDynCallThunks() {
   }
 }
 
-Function* Linker::generateImportThunk(Name name, const FunctionType* funcType) {
+Function* Linker::getImportThunk(Name name, const FunctionType* funcType) {
+  Name thunkName = std::string("__importThunk_") + name.c_str();
```

Comment:
Before, if the input code happened to contain a method with this prefix we would get an error when adding the second one to the module. But after this change, it will silently use that existing method as a thunk. There's a chance that would hit a validation error, but also it might not, and cause weird errors.

Hoisting the thunk prefix into a global and checking all input methods do not have that prefix seems safer. (Might also be worth renaming the prefix to something with `binaryen` in the name, so a conflict is really unlikely.)

An alternative might be to run `the duplicate-function-elimination` pass after s2wasm. It would coalesce all the duplicate thunks. We can do this in just 3 lines of code, but it would add time. But^2, it's a useful pass anyhow, and pretty fast.